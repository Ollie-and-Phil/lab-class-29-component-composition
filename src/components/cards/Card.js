import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ title, image }) => {
  return (
    <>
      {title && <header><h3>{title}</h3></header>}
      <figure>
        <img src={image} />
      </figure>
    </>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
};

export default Card;
