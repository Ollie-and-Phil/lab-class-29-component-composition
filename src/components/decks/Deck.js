import React from 'react';
import PropTypes from 'prop-types';

const Deck = ({ children }) => {
  return (
    <section>
      {children}
    </section>
  );
};

Deck.propTypes = {
  children: PropTypes.node.isRequired
};

export default Deck;
