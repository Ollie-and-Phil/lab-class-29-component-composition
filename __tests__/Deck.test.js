import React from 'react';
import { shallow } from 'enzyme';
import Deck from '../src/components/decks/Deck';
import Card from '../src/components/cards/Card';


describe('Deck component', () => {
  it('renders Deck', () => {
    const wrapper = shallow(<Deck><Card title="Phil" image="Phil.jpg" /></Deck>);
    expect(wrapper).toMatchSnapshot();
  });
});
