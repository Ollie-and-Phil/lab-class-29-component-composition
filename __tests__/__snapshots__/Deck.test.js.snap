// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck component renders Deck 1`] = `
<section>
  <Card
    image="Phil.jpg"
    title="Phil"
  />
</section>
`;
